src/Tempest/Database/src/Connection.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database;
+
+use PDOStatement;
+
+interface Connection
```

Comment:
I'm a little afraid the name will cause confusion:

We'd now have `DatabaseConnection` and `Connection`. `DatabaseConnection` is implemented by `MySqlConnection`, `SQLiteConnection`, …

`DatabaseConnection` was originally called `DatabaseDriver`, but I renamed it because `DatabaseConnection` seemed a better name from the user config point of view.

Could we come up with a better name for this new `Connection` interface? Or should we revert `DatabaseConnection` back to `DatabaseDriver`? 

@innocenzi @aidan-casey would appreciate your input as well

Reply:
I honestly can't think of a better name for `Connection` right now. `DatabaseDriver` seems perfectly fine instead of `DatabaseConnection`, but I get why you made the change. I think most people would understand the terminology and would have no problem with it in the config though.

Reply:
Oooh I have an even better idea: we can have an interface `DatabaseConfig`, and have `PostgresConfig`, etc implement it. Then we don't even need the environment one. 

This is definitely outside the scope of this PR. I'll create a new issue for it :)